demux/packet.c

```diff
@@ -209,6 +212,30 @@ size_t demux_packet_estimate_total_size(struct demux_packet *dp)
     if (dp->avpacket) {
         mp_assert(!dp->is_cached);
         size += ROUND_ALLOC(dp->len);
+        if (dp->is_wrapped_avframe) {
+            mp_require(dp->buffer);
+
+            const AVFrame *frame = (AVFrame *)dp->buffer;
+            if (frame->hw_frames_ctx) {
+                const AVHWFramesContext *hwctx = (AVHWFramesContext *)frame->hw_frames_ctx->data;
+                size += ROUND_ALLOC(av_image_get_buffer_size(hwctx->sw_format, hwctx->width,
```

Comment:
should guard against this returning -1

Reply:
added mp_require for that

Reply:
seems heavy-handed. just ignore it if there's an error?

Reply:
I would rather fix that if there are cases when this doesn't work. Alternative is that we will leak probably GB per second of memory and just crash anyway.